### DISCUSSION TITLE
Is longblob available?

### DISCUSSION THREAD

**Discussion by charleskhenry** [laravel/framework#36751] [Q&A] [answered]
I see that `$table->binary()` is available, but I don't see the matching type of `longblob`.

**Comment by charleskhenry** [comment, marked as answer]
I see this has already been patched [in March](https://github.com/laravel/framework/pull/32011).

Just use `$table->binary('field_name');`

**Reply by rodrigopedra** [reply]
Hey @charleskhenry just as a heads up, the PR you linked to was closed (rejected), not merged (accepted).

using `->binary()` still maps to a `blob` and not to `longblob`

https://github.com/laravel/framework/blob/90fbaa7ec53410508cd73c09e3f788e458dd8b92/src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php#L769-L778

One option is to use a macro to add an unsupported type to the `MySqlGrammar`

For example, in a migration file:

~~~php
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Schema\Grammars\MySqlGrammar;
use Illuminate\Support\Facades\Schema;
use Illuminate\Support\Fluent;

class CreateTestsTable extends Migration
{
    public function up()
    {
        // Add the macro to tell the grammar how to handle the unsupported type
        MySqlGrammar::macro('typeLongblob', function (Fluent $column) {
            return 'longblob';
        });
        
        Schema::create('tests', function (Blueprint $table) {
            $table->id();
            
            // call the low-level addColumn to use the unsupported type
            $table->addColumn('longblob', 'field_name');

            $table->timestamps();
        });
    }

    public function down()
    {
        Schema::dropIfExists('tests');
    }
}
~~~

If you need to use the `longblob` type across several migrations, you can add the macro to a Service Provider's `boot` method, so it is always available.
